packages/devtools_app/lib/src/primitives/notifications.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+class NotificationMessage {
+  NotificationMessage(
+    this.text, {
+    this.actions = const [],
+    this.duration = defaultDuration,
+  });
+
+  /// The default duration for notifications to show.
+  static const Duration defaultDuration = Duration(seconds: 7);
+
+  final String text;
+  final List<Widget> actions;
+  final Duration duration;
+}
+
+abstract class NotificationService {
+  /// Pushes a notification [message].
+  bool push(String message);
+  bool smartPush(
```

Comment:
new line between these methods and please add dart doc for `smartPush`.

and maybe rename to `pushNotification` so that its effect is a little more clear from the name it is from `smartPush`